src/dev/DynamicUrl.tsx

```diff
@@ -0,0 +1,69 @@
+import React, {
+  Dispatch,
+  FormEvent,
+  SetStateAction,
+  useEffect,
+  useRef,
+} from "react";
+import {
+  ButtonForm,
+  Curated,
+  DynamicUrlStyled,
+  ManualForm,
+} from "./DynamicUrl.styled";
+import { collections } from "./collections";
+
+interface DynamicUrlProps {
+  url: string;
+  setUrl: Dispatch<SetStateAction<string>>;
+}
+
+const DynamicUrl: React.FC<DynamicUrlProps> = ({ url, setUrl }) => {
```

Comment:
I don't suppose there's real world value in abstracting this component out to our Design Library, since it's more or less duplicated in Clover / Bloom?

Reply:
Could be... We may have more than just these two components at some point. Maybe once we hit three instances of this logic we look into it?